stream/src/main/scala/org/apache/pekko/stream/scaladsl/Source.scala

```diff
@@ -108,6 +108,21 @@ final class Source[+Out, +Mat](
     (mat, Source.fromPublisher(pub))
   }
 
+  /**
+   * Materializes this [[Source]] using the [[Sink]], immediately returning the values via the
+   * provided [[Sink]] as a new [[Source]].
+   *
+   * @param sink A sink which needs to materialize into a [[Future]], typically one
+   *             that collects values such as [[Sink.head]] or [[Sink.seq]]
+   * @return A new [[Source]] that contains the results of the provided [[Source]]'s
+   *         elements run with the [[Sink]]
+   * @since 1.2.0
+   */
+  def preMaterializeIntoSource[Mat2](sink: Graph[SinkShape[Out], Future[Mat2]]): Source[Mat2, Future[NotUsed]] =
+    Source.fromMaterializer { (mat, attr) =>
+      Source.future(this.withAttributes(attr).toMat(sink)(Keep.right).run()(mat))
```

Comment:
It's not using the `preMaterialize` but `fromMaterializer`, then the actual creation is deferred, the `preMaterializeIntoSource` name is not a right name.

Reply:
@mdedetrich I'm hoping to do a 1.2.0-M2 release in the next week or 2. Is PR in a position to be completed in that time frame?